hedera-node/src/main/java/com/hedera/services/store/contracts/HederaAbstractWorldUpdater.java

```diff
@@ -90,91 +109,183 @@ public HederaWorldUpdater updater() {
 	}
 
 	@Override
-	protected HederaWorldState wrappedWorldView() {
-		return world;
+	public Optional<WorldUpdater> parentUpdater() {
+		if (hederaWorldState instanceof WorldUpdater) {
+			return Optional.of((WorldUpdater) hederaWorldState);
+		} else {
+			return Optional.empty();
+		}
 	}
 
-	public static class HederaStackedUpdater
-			extends AbstractWorldUpdater<HederaAbstractWorldUpdater, HederaUpdateTrackingAccount> implements HederaWorldUpdater {
+	@Override
+	public void revert() {
+		for (int i = 0; i < numberOfAddressesClaimed; i++) {
+			hederaWorldState.reclaimContractId();
+		}
+		numberOfAddressesClaimed = 0;
+		createdAccounts.clear();
+		updatedAccounts.clear();
+		deletedAccounts.clear();
+	}
+
+	@Override
+	public Collection<HederaUpdateTrackingAccount> getTouchedAccounts() {
+		return updatedAccounts.values();
+	}
+
+	@Override
+	public Collection<Address> getDeletedAccountAddresses() {
+		return deletedAccounts;
+	}
+
+	public Collection<HederaWorldState.WorldStateAccount> getCreatedAccounts() {
+		return createdAccounts.values();
+	}
+
+	protected abstract HederaUpdateTrackingAccount getForMutation(Address address);
+
+	public static class HederaStackedUpdater extends HederaAbstractWorldUpdater implements HederaWorldUpdater {
+
+		private int numberOfAddressesClaimed = 0;
+
+		private final Map<Address, HederaWorldState.WorldStateAccount> createdAccounts = new HashMap<>();
```

Comment:
Why would the createdAccounts be `WorldStateAccounts` and not `HederaUpdateTrackingAcocunt`s?

Reply:
My understanding is that UpdateTrackingAccount tracks changes to existing accounts from the state. Since these are not yet created, we can use WorldStateAccount and set any properties we want to persist there.